### ISSUE TITLE
Set autoaway if open in some browser tab, but inactive for N minutes

### ISSUE THREAD

**Issue by benapetr** [thelounge/thelounge#2332]
It would be really nice if lounge could make me away even if it's open in some tab but I am not active in any session for N minutes, then auto set me online when I send or do something.

**Comment by Bebef** [comment]
Have you seen #1996 and #775 ?

**Comment by benapetr** [comment]
Probably not ;)

**Comment by benapetr** [comment]
#775 say "auto-away" when no clients are connected, I want auto-away even when clients are connected

#1996 suggests some improvements, but not what I suggested here.

**Comment by eskimo** [comment]
This would be nice. Came here looking for this exactly. For example, I like to just lock my screen and leave the house, but I like keeping tabs open so I don't have to reopen them when I get home. Would be nice if after a certain amount of time it assumed I was away.

**Comment by xPaw** [comment]
How do we differentiate when you're not using your PC vs you simply have TL open on a second screen and don't touch it?

**Comment by dgw** [comment]
Is it necessary to do so? The request is to mark the user as away if TL doesn't see any activity, not if the whole system doesn't.

**Comment by eskimo** [comment]
I agree with @dgw. Either way if you're not touching it you should be marked as away; be it locked, or just not interacting with the browser for whatever reason.

**Comment by xPaw** [comment]
Chrome devs are working on "idle detector", so that could come in handy when it ships: https://web.dev/idle-detection/

**Comment by jlaw90** [comment]
Just wanted to chime in here - I think we could detect when the tab loses focus (see https://stackoverflow.com/questions/2720658/how-to-detect-when-a-tab-is-focused-or-not-in-chrome-with-javascript, not sure if it's a cross-browser solution) and/or use the time of the last mouse/keyboard event (if the tab is active) to detect whether the user is active
